src/App.tsx

```diff
@@ -1,10 +1,52 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
-const USER_ID = 0;
+const USER_ID = 12005;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(Filter.All);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+
+  useEffect(() => {
+    setErrorMessage('');
+
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  useMemo(() => {
+    let copyTodos = todos;
```

Comment:
If you need to create a copy of array use it
```suggestion
    let copyTodos = [...todos];
```